File: Backend/TrackrAPI/TrackrAPI/Services/Catalogo/DominioDetalleService.cs
using TrackrAPI.Dtos.Catalogo;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Catalogo;
using System.Collections.Generic;

namespace TrackrAPI.Services.Catalogo
{
    public class DominioDetalleService
    {
        private IDominioDetalleRepository dominioDetalleRepository;
        private DominioDetalleValidatorService dominioDetalleValidatorService;

        public DominioDetalleService(IDominioDetalleRepository dominioDetalleRepository, DominioDetalleValidatorService dominioDetalleValidatorService)
        {
            this.dominioDetalleRepository = dominioDetalleRepository;
            this.dominioDetalleValidatorService = dominioDetalleValidatorService;
        }

        public DominioDetalleDto ConsultarDto(int idDominioDetalle)
        {
            var dominio = dominioDetalleRepository.ConsultarDto(idDominioDetalle);
            dominioDetalleValidatorService.ValidarExistencia(dominio);
            return dominio;
        }

        public IEnumerable<DominioDetalleGridDto> ConsultarPorDominio(int idDominio, int idCompania)
        {
            return dominioDetalleRepository.ConsultarPorDominio(idDominio, idCompania);
        }

        public void Agregar(DominioDetalle dominioDetalle)
        {
            dominioDetalleValidatorService.ValidarAgregar(dominioDetalle);
            dominioDetalle.Habilitado = true;
            dominioDetalleRepository.Agregar(dominioDetalle);
        }

        public void Editar(DominioDetalle dominioDetalle)
        {
            dominioDetalle.Habilitado = true;
            dominioDetalleValidatorService.ValidarEditar(dominioDetalle);
            dominioDetalleRepository.Editar(dominioDetalle);
        }

        public void Eliminar(int idDominioDetalle)
        {
            var dominioDetalle = dominioDetalleRepository.Consultar(idDominioDetalle);
            dominioDetalleValidatorService.ValidarEliminar(idDominioDetalle);

            // dominioDetalleRepository.Eliminar(dominioDetalle);

            dominioDetalle.Habilitado = false;
            dominioDetalleRepository.Editar(dominioDetalle);
        }
    }
}

File: Frontend/trackr-administrador/src/app/shared/http/catalogo/dominio-detalle.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { DominioDetalle } from '@models/catalogo/dominio-detalle';

@Injectable()
export class DominioDetalleService {
  private dataUrl = 'dominioDetalle/';

  constructor(public http: HttpClient) { }

  public consultar(idDominioDetalle: number): Observable<DominioDetalle> {
    return this.http.get<DominioDetalle>(this.dataUrl + `consultar/${idDominioDetalle}`);
  }

  public consultarPorDominio(idDominio: number): Observable<DominioDetalle[]> {
    return this.http.get<DominioDetalle[]>(this.dataUrl + `consultarPorDominio/${idDominio}`);
  }

  public agregar(dominioDetalle: DominioDetalle): Observable<void> {
    return this.http.post<void>(this.dataUrl + 'agregar', dominioDetalle);
  }

  public editar(dominioDetale: DominioDetalle): Observable<void> {
    return this.http.put<void>(this.dataUrl + 'editar', dominioDetale);
  }

  public eliminar(idDominioDetalle: number): Observable<void> {
    return this.http.delete<void>(this.dataUrl + `eliminar/${idDominioDetalle}`);
  }

}
